includes/class-freemius.php

```diff
@@ -10919,6 +11387,50 @@ function get_site() {
             return $this->_site;
         }
 
+        /**
+         * @author Leo Fajardo (@leorw)
+         * @since 2.4.3
+         */
+        function store_site( $site ) {
+            $this->_site = $site;
+            $this->_store_site( true );
+        }
+
+        /**
+         * @author Leo Fajardo (@leorw)
+         * @since 2.4.3
+         */
+        private function delete_current_install( $back_up ) {
+            // Back up and delete the unique ID.
+            if ( $back_up ) {
+                self::$_accounts->set_option( 'prev_unique_id', $this->get_anonymous_id() );
+            }
+
+            self::$_accounts->set_option( 'unique_id', null );
+
+            if ( $back_up ) {
+                // Back up and delete the current install.

```

Comment:
@fajardoleo it's not clear why you need to "backup" the site, isn't the site was loaded from the DB already? why would we need to store it again. Also, the comment is incorrect, this line doesn't delete the install.

Reply:
@vovafeldman We back up a site so that we can restore it if the automatic resolution attempt fails (during an automatic resolution attempt, we delete the site).

Reply:
@fajardoleo in that case please update the comment accordingly. Also, add documentation to the method that explains the potential backup and reasoning.